server/comments.js

```diff
@@ -35,6 +35,51 @@ function groupByArea(deltas) {
 	});
 }
 
+function totalDeltasForArea(areaDelta, delta) {
+	if (!areaDelta) {
+		return {};
+	}
+
+	// Produce an array of arrays:
+	// [ [ chunks in use in first commit ] , [ chunks in use in second commit ] ]
+	// The items will be unique inside each array.
+	const chunksInUse = ['firstChunks', 'secondChunks']
+		.map(chunkType => areaDelta.reduce(
+			(acc, group) => {
+				for (chunk of group[chunkType]) {
+					acc.add(chunk);
+				}
+				return acc;
+			},
+			new Set()
+		))
+		.map(set => [...set]);
+
+	// Produce an array of size objects, representing the sum of all the chunks for each commit:
+	// [ { stat_size: 0, parsed_size: 0, gzip_size: 0 }, { stat_size: 0, parsed_size: 0, gzip_size: 0 } ]
+	// The first object is for the first commit, and the second object for the second commit.
+	const chunkSizes = ['firstSizes', 'secondSizes']
+		.map((property, index) => chunksInUse[index].reduce(
+			(acc, chunkName) => {
+				const chunk = delta.chunks.find(chunk => chunk.name === chunkName) || {};
+				for (size in chunk[property]) {
+					acc[size] = chunk[property][size] + (acc[size] || 0);
+				}
+				return acc;
+			},
+			{}
```

Comment:
Is it possible that the `chunksInUse[ index ]` array is empty for a particular area? Then the `reduce` would loop over an empty array and would return an empty object rather than `{ stat_size: 0, parsed_size: 0, gzip_size: 0 }`.

It might be better to do the `reduce` with a `{ stat_size: 0, parsed_size: 0, gzip_size: 0 }` initial value.

The `delta.js` module has a nice `sumSizesOf` helper to sum two size objects together.